### PR TITLE
Split MarkV into multiple files

### PR DESCRIPTION
**Pull request by dj2**
This CL breaks the monolithic markv_codec file into files for the base
class, encoder, decoder and logger.

### CODE REVIEW

**Comment by dj2** [comment]
@dneto0 @alan-baker ping

File: source/comp/markv.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2018 Google Inc.
```
Comment (alan-baker) [inline source/comp/markv.cpp]: The company should be Google LLC.

Comment (dj2) [reply source/comp/markv.cpp]:
```
Development/spirv-tools/SPIRV-Tools % gg "Google LLC"|wc -l
138
Development/spirv-tools/SPIRV-Tools % gg "Google Inc"|wc -l
298
```

Comment (dj2) [reply source/comp/markv.cpp]: Looks like a larger cleanup is needed here.

**Pull request merged**